pynio/block.py

```diff
@@ -31,7 +31,7 @@ def save(self):
         config['name'] = self._name
         config['type'] = self._type
         # load template and then reload config
-        self.template = self._instance.blocks_types[self._type].template
+        self.template = self._instance.blocks_types[self._type]
```

Comment:
But blocks_types are just  readonly dictionaries.

I know why it didn't catch it.  It's because the template is hardly used.
It should be gotten from the instance anyway.  I'll fix it tomorrow
On Feb 18, 2015 7:31 PM, "Chris Brackert" <notifications@github.com> wrote:

> In pynio/block.py
> <https://github.com/neutralio/pynio/pull/19#discussion_r24962139>:
>
> > @@ -31,7 +31,7 @@ def save(self):
> >          config['name'] = self._name
> >          config['type'] = self._type
> >          # load template and then reload config
> > -        self.template = self._instance.blocks_types[self._type].template
> > +        self.template = self._instance.blocks_types[self._type]
>
> This doesn't make sense. setting self.template to a Block? Isn't
> self.template supposed to be a Properties object?
>
> It also concerns me that no unit tests caught this.
>
> Not to mention, I think a minor refactoring task should happen at some
> point so that blocks_types are not Block objects but instead BlockType
> objects.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/neutralio/pynio/pull/19/files#r24962139>.
>